src/Kestrel.Transport.Abstractions/Internal/TransportConnection.Features.cs

```diff
@@ -160,18 +199,30 @@ IDuplexPipe IConnectionTransportFeature.Application
 
         TFeature IFeatureCollection.Get<TFeature>()
         {
-            if (typeof(TFeature) == typeof(IHttpConnectionFeature))
+            if (typeof(TFeature) == IHttpConnectionFeatureType)
```

Comment:
The generic version should use `typeof` for the Jit so its `T` == `T`
`typeof(TFeature) == typeof(IHttpConnectionFeature)`

Reply:
`Type` version is different as its `variable == `

Reply:
Good to know

Reply:
I'll fix this in a subsequent commit.